.circleci/workflows.yml

```diff
@@ -1997,13 +2061,58 @@ jobs:
     resource_class: << parameters.resource_class >>
     steps:
       - restore_cached_workspace
-      - build-binary
+      - check-if-binary-exists
+      - build-and-package-binary
       - build-cypress-npm-package:
           executor: << parameters.executor >>
-      - verify_should_persist_artifacts
       - upload-build-artifacts
+      - verify_should_persist_artifacts
       - post-install-comment
 
+  create-build-artifacts-and-trigger-publish:
+    <<: *defaults
+    parameters:
+      <<: *defaultsParameters
+      resource_class:
+        type: string
+        default: xlarge
+    resource_class: << parameters.resource_class >>
+    steps:
+      - restore_cached_workspace
+      - check-if-binary-exists
+      - build-binary
+      - trigger-publish-binary-pipeline
+
+  get-published-binary:
+    <<: *defaults
+    parameters:
+      <<: *defaultsParameters
+      resource_class:
+        type: string
+        default: large
+    resource_class: << parameters.resource_class >>
+    steps:
+      - restore_cached_workspace
+      - run:
+          name: Check pipeline info
+          command: cat ~/triggered_pipeline.json
+      - run:
+          name: Download binary artifacts
+          command: |
+            source ./scripts/ensure-node.sh
+            node ./scripts/binary/get-published-artifacts.js --pipelineInfo ~/triggered_pipeline.json
+      - persist_to_workspace:
+          root: ~/
+          paths:
+            - cypress/cypress.zip
+            - cypress/cypress.tgz
+            - cypress/binary-url.json
+            - cypress/npm-package-url.json
```

Comment:
Would it be better to have a single `artifact_urls.json` file that includes both of these?

Reply:
yeah, maybe. I didn't really want to change the way we do it now which is to store them each in their own file 🤷🏻‍♂️

Reply:
Ah, I thought this was new. I'd say leave it as is.